### PR TITLE
Adding router configs into cloudRouterFactory

### PR DESCRIPTION
**Pull request by SophieGuo410**

### CODE REVIEW

File: ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java [hunk 1]
```diff
@@ -534,6 +536,20 @@
   @Default("com.github.ambry.router.OperationController")
   public final String operationController;
 
+  /**
+   * Implementation class for StoreKeyConverterFactory
```
Comment (cgtz) [inline ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java:540]: Could you mention that these configs are specific to the embedded AmbryRequests in cloud router only

Comment (SophieGuo410) [reply ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java:540]: Sure, updated.

File: ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java [hunk 2]
```diff
@@ -650,5 +668,8 @@ public RouterConfig(VerifiableProperties verifiableProperties) {
         verifiableProperties.getBoolean(ROUTER_PUT_REQUEST_USE_JAVA_NATIVE_CRC32, false);
     operationController =
         verifiableProperties.getString(OPERATION_CONTROLLER, "com.github.ambry.router.OperationController");
+    routerRequestHandlerNumOfThreads = verifiableProperties.getInt(ROUTER_REQUEST_HANDLER_NUM_OF_THREADS, 7);
+    routerStoreKeyConverterFactory = verifiableProperties.getString(ROUTER_REQUEST_HANDLER_NUM_OF_THREADS,
+        "com.github.ambry.store.StoreKeyConverterFactoryImpl");
   }
```
Comment (JingQianCloud) [inline ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java:674]: Can we add one unit test to verify, the thread number and StoreKeyConverter are used as specified by the verifiableProperties.ROUTER_REQUEST_HANDLER_NUM_OF_THREADS and verifiableProperties.ROUTER_STORE_KEY_CONVERTER_FACTORY.

Not sure if it's easy. If it's not, manual test is ok too.

File: ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java [hunk 3]
```diff
@@ -650,5 +668,8 @@ public RouterConfig(VerifiableProperties verifiableProperties) {
         verifiableProperties.getBoolean(ROUTER_PUT_REQUEST_USE_JAVA_NATIVE_CRC32, false);
     operationController =
         verifiableProperties.getString(OPERATION_CONTROLLER, "com.github.ambry.router.OperationController");
+    routerRequestHandlerNumOfThreads = verifiableProperties.getInt(ROUTER_REQUEST_HANDLER_NUM_OF_THREADS, 7);
+    routerStoreKeyConverterFactory = verifiableProperties.getString(ROUTER_REQUEST_HANDLER_NUM_OF_THREADS,
```
[the remark this answers is not in the document]
Comment (SophieGuo410) [reply ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java]: Thanks for catching that. updated.

File: ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java (hunk 2 above)
Comment (SophieGuo410) [reply ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java:674]: I add a test to my local and test it. but I feel it's not necessary to add to the central test list.

**Review by JingQianCloud — APPROVED** [review]

**Review by cgtz — APPROVED** [review]

**Pull request merged**